authz/providers/azure/rbac/rbac.go

```diff
@@ -427,7 +428,7 @@ func buildCheckAccessURL(base url.URL, resourceID string, hasNamespace bool, nam
 	return base
 }
 
-func (a *AccessInfo) sendCheckAccessRequest(ctx context.Context, checkAccessUsername string, checkAccessURL url.URL, checkAccessBody *CheckAccessRequest, ch chan *authzv1.SubjectAccessReviewStatus) error {
+func (a *AccessInfo) sendCheckAccessRequest(ctx context.Context, checkAccessUsername string, checkAccessURL url.URL, checkAccessBody *CheckAccessRequest, ch chan *authzv1.SubjectAccessReviewStatus, isTrackedResource bool) error {
```

Comment:
do we really need this argument 'isTrackedResource bool'? 

For a non-tracked child resource, regardless its existence,  'az role assignment list ...' will return an empty array of role assignment array if there is any. 

For a tracked child resource that ARM knows of,  'az role assignment list ...' will return 'ResourceNotFound' if this tracked child resource doesn't exist, and return a role assignment array (if there is any) for this resource if it exists.

If you have this additional, when we switch the routing type of managedClusters/managedNamespaces between 'proxyonly' and 'tracked', we always need to update the guard logic here.
![image](https://github.com/user-attachments/assets/b187bbfb-5466-4905-ae6a-da81f21c92d8)